sandbox/src/main/java/org/robolectric/internal/bytecode/ShadowWrangler.java

```diff
@@ -366,6 +370,36 @@ private Method findShadowMethodDeclaredOnClass(
           // Found a looseSignatures match, but continue looking for an exact match.
           foundMethod = method;
         }
+      } else {
+        // Or maybe support @ClassName.
+        Annotation[][] annotations = method.getParameterAnnotations();
```

Comment:
I was under the impression that the return type is not as important when doing loose signatures.

For instance, Java will not let you override a method when only the return type is different. 

I can take a closer look at this tomorrow.

Reply:
the current `if (isLooseSignature)` branch looks like not checking function return type.

If we want to make sure function return type also matches, is it ok to add one more return-type in following functions?
```java
ShadowWrangler.pickShadowMethod(Class<?> definingClass, String name, Class<?>[] paramTypes, @Nullable Class<?> rType)
ShadowWrangler.findShadowMethod(..., @Nullable Class<?> rType)
ShadowWrangler.findShadowMethodDeclaredOnClass(..., @Nullable Class<?> rType)
```

Reply:
@utzcoz is there a specific method you had in mind that may cause problems?

Reply:
Also, the method where this stuff is implemented, [ShadowWrangler.findShadowMethodDeclaredOnClass](https://github.com/robolectric/robolectric/blob/master/sandbox/src/main/java/org/robolectric/internal/bytecode/ShadowWrangler.java#L334-L335), does not pass in the return type. I am looking into seeing what would break if we also validation the return type.

Reply:
@utzcoz @Mia0451 replacing a couple of looseSignatures SGTM.

Can you start with this one:

https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowStaticLayout.java

The plan was to create a spreadsheet of looseSignatures usages and do them one-by-one. There are probably some that do not need looseSignatures/ClassName anymore (i.e. they were added for a new type in Q, but now Q is the minimum required compileSdk).

Reply:
done.
The value used in @ClassName for the parameters in ShadowStaticLayout.java followed these two 
https://github.com/AndroidSDKSources/android-sdk-sources-for-api-level-23/blob/master/android/text/StaticLayout.java
https://github.com/AndroidSDKSources/android-sdk-sources-for-api-level-28/blob/master/android/text/StaticLayout.java

Reply:
test is failing. looks like I also need to add `@ClassName` parsing in `SdkStore$ClassInfo.findMethod`?